integration-cli/docker_cli_pull_test.go

```diff
@@ -285,5 +285,11 @@ func (s *DockerCLIPullSuite) TestPullLinuxImageFailsOnWindows(c *testing.T) {
 func (s *DockerCLIPullSuite) TestPullWindowsImageFailsOnLinux(c *testing.T) {
 	testRequires(c, DaemonIsLinux, Network)
 	_, _, err := dockerCmdWithError("pull", "mcr.microsoft.com/windows/servercore:ltsc2022")
-	assert.ErrorContains(c, err, "no matching manifest for linux")
+
+	errorMessage := "no matching manifest for linux"
```

Comment:
Also thinking of some things to consider; honestly, the containerd error is less informative than the old one, because it doesn't show _WHAT_ we weren't able to match (which _could_ be obvious if a user explicitly passed a `--platform`, but less so if not, but could also be that user passed a _partial_ platform, and we normalized it).

The "non-snapshotter" error looks to be this one; https://github.com/moby/moby/blob/0d9da7367dd79e6e5c46ef4a37ca0fbe79d2bf8f/distribution/pull_v2.go#L939-L945

For containerd, the error is produced here; https://github.com/moby/moby/blob/0d9da7367dd79e6e5c46ef4a37ca0fbe79d2bf8f/vendor/github.com/containerd/containerd/images/handlers.go#L310-L312


I guess _ideally_ the containerd error would be somewhat more informative (_what_ were we trying to match, and couldn't find?), but it looks like that's less trivial, because that's fully up to the `HandlerFunc` that's passed (which does the filtering).

The alternative here would be to handle this on our side: when we get the result. The error returned is typed; it's a `errdefs.ErrNotFound`, so we can decorate the error (and perhaps we can decorate it regardless of what the reason was), i.e.;

```go
if cerrdefs.IsNotFound(err) {
    err = fmt.Errorf("no matching manifest for %s: %w", formatPlatform(e.platform), err)
} else {
    err = fmt.Errorf("failed to find matching manifest for %s: %w", formatPlatform(e.platform), err)
}
```

(I removed "in the manifest list entries" from the above, but maybe we should keep it, not sure if it's "too much")

Reply:
- opened https://github.com/moby/moby/issues/46765 for the above, as it's not related to the test